src/storage/src/hummock/utils.rs

```diff
@@ -124,17 +125,40 @@ pub(crate) fn search_sst_idx(ssts: &[SstableInfo], key: UserKey<&[u8]>) -> usize
 /// a specific table id. Returns the sst ids after pruning.
 pub fn prune_overlapping_ssts<'a, R, B>(
     ssts: &'a [SstableInfo],
+    indices: &'a [usize],
     table_id: TableId,
     table_key_range: &'a R,
 ) -> impl DoubleEndedIterator<Item = &'a SstableInfo>
 where
     R: RangeBounds<TableKey<B>>,
     B: AsRef<[u8]> + EmptySliceRef,
 {
-    ssts.iter()
+    indices
+        .iter()
+        .map(|index| &ssts[*index])
         .filter(move |info| filter_single_sst(info, table_id, table_key_range))
 }
 
+/// Prune overlapping SSTs that does not overlap with a specific key range or does not overlap with
+/// a specific table id. Returns the sst ids after pruning with reverse order in a collection.
+pub fn prune_overlapping_ssts_rev<'a, R, B>(
```

Comment:
`rev` can be done outside easily so I think we don't need this method.

Reply:
> `rev` can be done outside easily so I think we don't need this method.

fixed